src/packagedcode/go_mod.py

```diff
@@ -31,10 +30,48 @@
 import re
 
 import attr
+from packageurl import PackageURL
+
 
+TRACE = False
 
+logger = logging.getLogger(__name__)
+
+if TRACE:
+    import sys
+    logging.basicConfig(stream=sys.stdout)
+    logger.setLevel(logging.DEBUG)
+
+
+@attr.s()
+class GoModules(object):
+    namespace = attr.ib(default=None)
+    name = attr.ib(default=None)
+    version = attr.ib(default=None)
+    module = attr.ib(default=None)
+
+    @property
+    def purl_gosum(self):
```

Comment:
How is this different from purl_gomod? IMHO you will get the same whether or not you provide a version

Reply:
We don't need version number in `purl` because it was mentioned in the go modules that the version number which is written in the go.mod file is the minimum version required. But we need version number for requirements field. So I made it like this.

Reply:
ok, but if you where to use one function with an argument to `include_version=True` or `False`, you could have a single function and a bit less code?

Reply:
I added new code. Please check, in that the way you were asking me to do it?